transaction-pool/src/lib.rs

```diff
@@ -0,0 +1,153 @@
+use solana_sdk::pubkey::Pubkey;
+use std::collections::BTreeMap;
+use std::collections::HashMap;
+use std::collections::VecDeque;
+
+#[derive(Default, Clone, PartialEq, Debug)]
+pub struct Item {
+    pub priority: u64,
+    pub units: u64,
+    pub time: u64,
+    pub id: (u32, u32),
+}
+
+#[derive(Default, Clone)]
+pub struct Pool {
+    pub max_age: u64,
+    pub max_bucket_cu: u64,
+    pub max_block_cu: u64,
+    pub count: usize,
+    items: BTreeMap<u64, VecDeque<Item>>,
```

Comment:
Assuming `Item`s in the Pool are `Packet` not `PacketBatch`? Perhaps `SigVerify` receives and sends `Packet`s?

Reply:
Yea, these are just pointers, u32 id for the PacketBatch and u32 for the index in the packet.